### PR TITLE
[agent][node-analyzer][sysdig-deploy] Helm unit tests to check apiEnd…

### PR DESCRIPTION
**Pull request by yashgiri**
…point and CollectorEndpoint values for different regions

## What this PR does / why we need it:
Added helm tests to check apiEndpoint and CollectorEndpoint values that are set depending on the region supplied.
## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with chart name (e.g. [mychartname])
- [x] Chart Version bumped for the respective charts
- [ ] Changelog updated for the charts with version updates.
- [ ] Variables are documented in the README.md (or README.tpl in some charts)
- [ ] Check GithubAction checks (like lint) to avoid merge-check stoppers

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

**Comment by yashgiri** [comment]
Are there any changes you guys would recommend adding to the readme?
I used the helm unit testing from here : https://github.com/quintush/helm-unittest

**Comment by yashgiri** [comment]
Here is the result of the tests running
```
yash@yash charts % helm unittest --helm3 ./node-analyzer               

Chart [ node-analyzer ] ./node-analyzer

 PASS  Test benchmark runner collector_endpoint value for all regions	node-analyzer/tests/collector-endpoint-region_test.yaml

Charts:      1 passed, 1 total
Test Suites: 1 passed, 1 total
Tests:       6 passed, 6 total
Snapshot:    0 passed, 0 total
Time:        121.200942ms

yash@yash charts % helm unittest --helm3 ./agent        

Chart [ agent ] ./agent

 PASS  Test agent configmap collector value set by api_endpoint for all regions (us1,us2,us3,us4,eu1,au1)	agent/tests/api-endpoint-region_test.yaml

Charts:      1 passed, 1 total
Test Suites: 1 passed, 1 total
Tests:       6 passed, 6 total
Snapshot:    0 passed, 0 total
Time:        97.502689ms
```

**Review by ironashram — APPROVED** [review]
LGTM!

**Comment by yashgiri** [comment]
Hi @ironashram, I and @aroberts87 were discussing how there are 2 tests folder in the agent sub chart. One of the tests folder (agent/tests) is the one I included along with the framework I mentioned: https://github.com/quintush/helm-unittest. And the other one is the folder written by Ashwin (agent/templates/tests) before he left for chart upgrade testing. these tests don't really clash and technically I can include my files in Ashwin's folder but it will have to make me run my unit tests always with the a -f command which might make things inconsistent. The other option is to see if we can rename the folder under templates as it doesn't look like it should break things. Ashwin's PR that created that folder is here - https://github.com/sysdiglabs/charts/commit/6d5d81b8435a9bb4be39f3e6566692064fca46fd

**Comment by ironashram** [comment]
> Hi @ironashram, I and @aroberts87 were discussing how there are 2 tests folder in the agent sub chart. One of the tests folder (agent/tests) is the one I included along with the framework I mentioned: https://github.com/quintush/helm-unittest. And the other one is the folder written by Ashwin (agent/templates/tests) before he left for chart upgrade testing. these tests don't really clash and technically I can include my files in Ashwin's folder but it will have to make me run my unit tests always with the a -f command which might make things inconsistent. The other option is to see if we can rename the folder under templates as it doesn't look like it should break things. Ashwin's PR that created that folder is here - [6d5d81b](https://github.com/sysdiglabs/charts/commit/6d5d81b8435a9bb4be39f3e6566692064fca46fd)

i guess these 2 type of tests are different things and as such they should coexist without issues, i do not have evidence of what happens if you rename the folder tho

**Comment by aroberts87** [comment]
> > Hi @ironashram, I and @aroberts87 were discussing how there are 2 tests folder in the agent sub chart. One of the tests folder (agent/tests) is the one I included along with the framework I mentioned: https://github.com/quintush/helm-unittest. And the other one is the folder written by Ashwin (agent/templates/tests) before he left for chart upgrade testing. these tests don't really clash and technically I can include my files in Ashwin's folder but it will have to make me run my unit tests always with the a -f command which might make things inconsistent. The other option is to see if we can rename the folder under templates as it doesn't look like it should break things. Ashwin's PR that created that folder is here - [6d5d81b](https://github.com/sysdiglabs/charts/commit/6d5d81b8435a9bb4be39f3e6566692064fca46fd)
> 
> i guess these 2 type of tests are different things and as such they should coexist without issues, i do not have evidence of what happens if you rename the folder tho

Sounds good, @ironashram. I just had a small concern about having multiple test frameworks for the same chart and wasn't sure if you were aware of the existence of the other one yet so I asked @yashgiri to hold off merging until you had a chance to reply. @yashgiri - we can go ahead and merge this now once the conflicts are resolved. 👍🏻

**Review by aroberts87 — APPROVED** [review]

**Review by lilx1ao — APPROVED** [review]

**Comment by yashgiri** [comment]
Thank you for the review.

File: charts/agent/tests/api_endpoint_region_test.yaml
Comment (aroberts87) [inline charts/agent/tests/api_endpoint_region_test.yaml:66]: just for my own curiosity, why do we have `asserts.matchRegex.path` here set to `data.dragent\.yaml`, but in the node analyzer test we use `asserts.equal.path` and `asserts.equal.value`? The method used in the node analyzer check seems to be more specific and clear about what it's checking. Should they be made consistent?

File: charts/node-analyzer/tests/collector_endpoint_region_test.yaml
Comment (aroberts87) [inline charts/node-analyzer/tests/collector_endpoint_region_test.yaml]: nit: adding a newline here would appease GitHub. 🙂

Comment (yashgiri) [reply charts/node-analyzer/tests/collector_endpoint_region_test.yaml]: got it

File: charts/agent/tests/api_endpoint_region_test.yaml
Comment (yashgiri) [reply charts/agent/tests/api_endpoint_region_test.yaml:66]: Oh yeah, this one was pretty annoying to deal with but the value inside data.dragent.yaml where is as follows
```
data:
  dragent.yaml: |
    new_k8s: true
    k8s_cluster_name: yash-k8s
    security:
      enabled: true
    commandlines_capture:
      enabled: true
    memdump:
      enabled: true

    collector: collector.sysdigcloud.com
    k8s_coldstart:
      max_parallel_cold_starts: 1
```
since the whole value is a block of values because of the "|" present above, matchRegex was a better way of verifying it.

Comment (aroberts87) [reply charts/agent/tests/api_endpoint_region_test.yaml:66]: ah, gotcha. sounds good! thanks for explaining that.

**Review by aroberts87 — APPROVED** [review]

**Pull request merged**
